docs/blocks/transform.md

```diff
@@ -1 +1,170 @@
+   {doc}`../builtin/range`) and {py:class}`rasterix.RasterIndex` (see
+   {doc}`../earth/raster`).
+
+## Example
+
+As a real-world example, let's create a custom
+{py:class}`xarray.indexes.CoordinateTransform` that wraps an
+{py:class}`astropy.wcs.WCS` object. This Xarray coordinate transform adapter
+class simply maps Xarray dimension and coordinate variable names to pixel
+and world axis names of the [shared Python interface for World Coordinate
+Systems](https://doi.org/10.5281/zenodo.1188874) used in Astropy.
+
+```{note}
+This example is taken and adapted from
+[this gist](https://gist.github.com/Cadair/4a03750868e044ac4bdd6f3a04ed7abc) by
+Stuart Mumford.
+
+It only provides basic integration between Astropy's WCS and Xarray
+coordinate transforms. More advanced integration could leverage the
+[slicing capabilities of WCS objects](https://docs.astropy.org/en/latest/wcs/wcsapi.html#slicing-of-wcs-objects)
+in a custom {py:class}`xarray.indexes.CoordinateTransformIndex` subclass.
+
+```
+
+```{code-cell} python
+from collections.abc import Hashable
+from typing import Any
+
+import numpy as np
+import xarray as xr
+from astropy.wcs import WCS
+
+
+def escape(name):
+    return name.replace(".", "_").replace("custom:", "")
+
+
+class WCSCoordinateTransform(xr.indexes.CoordinateTransform):
+    """Lightweight adapter class for the World Coordinate Systems (WCS) API.
+
+    More info: https://docs.astropy.org/en/latest/wcs/wcsapi.html
+
+    """
+    def __init__(
+        self,
+        wcs: WCS,
+    ):
+        pixel_axis_names = [
+            pan or f"dim{i}"
+            for i, pan in enumerate(wcs.pixel_axis_names)
+        ]
+        world_axis_names = [
+            escape(wan or wphy or f"coord{i}")
+            for i, (wan, wphy) in enumerate(
+                zip(wcs.world_axis_names, wcs.world_axis_physical_types)
+            )
+        ]
+        dim_size = {
+            name: size
+            for name, size in zip(pixel_axis_names, wcs.array_shape)
+        }
+
+        super().__init__(world_axis_names, dim_size, dtype=np.dtype(float))
+        self.wcs = wcs
+
+    def forward(self, dim_positions: dict[str, Any]) -> dict[Hashable, Any]:
+        """Perform array -> world coordinate transformation."""
+
+        pixel = [dim_positions[dim] for dim in self.dims]
+        world = self.wcs.array_index_to_world_values(*pixel)
+
+        return {name: w for name, w in zip(self.coord_names, world)}
+
+    def reverse(self, coord_labels: dict[Hashable, Any]) -> dict[str, Any]:
+        """Perform world -> array coordinate reverse transformation."""
+
+        world = [coord_labels[name] for name in self.coord_names]
+        pixel = self.wcs.world_to_array_index_values(*world)
+
+        return {name: p for name, p in zip(self.dims, pixel)}
+
+
+```
+
+```{code-cell} python
+---
+tags: [remove-cell]
+---
+%xmode minimal
+xr.set_options(
+    display_expand_indexes=True,
+    display_expand_attrs=False,
```

Comment:
```suggestion
    display_expand_data=False,
    display_expand_attrs=False,
```

Reply:
I set `display_expand_data=True` for showing the lazy coordinate data repr.

Reply:
What I was intending is to avoid the "wall of floats" in the repr - especially for da_3d - that I find distracting.  I guess another option is to make them datasets? Or in some of the tutorial notebooks we set `np.set_printoptions(threshold=10, edgeitems=2)` (https://tutorial.xarray.dev/advanced/apply_ufunc/simple_numpy_apply_ufunc.html)

Reply:
Yeah it'd be great if there was some way of inserting a preamble in all files.